cmake/modules/SIMDExt.cmake

```diff
@@ -34,28 +35,44 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
       return ret;
     }
     int main() { foo(0); }" HAVE_ARMV8_CRYPTO)
+
   set(CMAKE_REQUIRED_QUIET ${save_quiet})
   if(HAVE_ARMV8_CRC)
     message(STATUS " aarch64 crc extensions supported")
   endif()
+
   if(HAVE_ARMV8_CRYPTO)
     message(STATUS " aarch64 crypto extensions supported")
   endif()
-  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH)
+
+  # don't believe only the -march support; gcc 4.8.5 on RHEL/CentOS says
+  # it supports +crc but hasn't got the intrinsics or arm_acle.h.  Test for
+  # the actual presence of one of the intrinsic functions.
+  if(HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH)
+    check_cxx_source_compiles("
+      #include <inttypes.h>
+      int main() { uint32_t a; uint8_t b; __builtin_aarch64_crc32b(a, b); }
+    " HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+  endif()
```

Comment:
>  i dunno what happens if you try to refer to the intrinsic but do not pass crc+crypto.

the intrinsics are implemented using the the GCC builtin functions like `__builtin_aarch64_crc32b()` which is directly mapped to the corresponding assembly without the help of the `-m<machine>`. (i skimmed https://github.com/gcc-mirror/gcc/blob/master/gcc/config/aarch64/aarch64-builtins.c, and got this conclusion =))

>  so i guess we can just drop CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH)

okay, we cannot drop it, as we are trying to set `ARMV8_CRC_COMPILE_FLAGS` if `-march=armv8-a+crc+crypto` is supported. so maybe we can s/HAVE_ARMV8_CRC_CRYPTO_INTRINSICS/HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH/` at line 59. and maybe rename `HAVE_ARMV8_CRC_CRYPTO_INTRINSICS_MARCH` to `HAVE_ARMV8_CRC_CRYPTO_MARCH`.